File: go/space-age/space_age.go
package space

// Planet is a wrapper type over a string represent a planets' name
type Planet string

// Age accepts an age in seconds (float64) and returns their age on
// the given Planet. If an invalid Planet is given, it returns -1.00
func Age(seconds float64, planet Planet) float64 {
	switch planet {
	case "Mercury":
		return (seconds / 31557600) / 0.2408467
	case "Venus":
		return (seconds / 31557600) / 0.61519726
	case "Earth":
		return seconds / 31557600
	case "Mars":
		return (seconds / 31557600) / 1.8808158
	case "Jupiter":
		return (seconds / 31557600) / 11.862615
	case "Saturn":
		return (seconds / 31557600) / 29.447498
	case "Uranus":
		return (seconds / 31557600) / 84.016846
	case "Neptune":
		return (seconds / 31557600) / 164.79132
	default:
		return -1.00
	}
}

File: erlang/space-age/src/space_age.erl
-module(space_age).

-export([age/2]).

earth_divisor() -> #{
    mercury => 0.2408467,
    venus =>  0.61519726,
    mars =>  1.8808158,
    jupiter =>  11.862615,
    saturn =>  29.447498,
    uranus =>  84.016846,
    neptune => 164.79132
}.

seconds_to_years(seconds) -> seconds / 31557600.

age(_Planet, _Seconds) -> undefined.
